src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Bjorhus.hpp

```diff
@@ -0,0 +1,287 @@
+#include "Evolution/Systems/GeneralizedHarmonic/ConstraintDamping/Tags.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/AnalyticData/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/AnalyticSolution.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace domain::Tags {
+template <size_t Dim, typename Frame>
+struct Coordinates;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace GeneralizedHarmonic::BoundaryConditions::detail {
+enum class ConstraintPreservingBjorhusType {
+  ConstraintPreserving,
+  ConstraintPreservingPhysical
+};
+
+ConstraintPreservingBjorhusType
+convert_constraint_preserving_bjorhus_type_from_yaml(
+    const Options::Option& options);
+}  // namespace GeneralizedHarmonic::BoundaryConditions::detail
+
+namespace GeneralizedHarmonic::BoundaryConditions {
+/*!
+ * \brief Sets constraint preserving boundary conditions using the Bjorhus
+ * method.
+ *
+ * \details Boundary conditions for the generalized harmonic evolution system
+ * can be divided in to three parts, constraint-preserving, physical and gauge
+ * boundary conditions.
+ *
+ * The generalized harmonic (GH) evolution system is a first-order reduction of
+ * Einstein equations brought about by the imposition of GH gauge. This
+ * introduces constraints on the free (evolved) variables in addition to the
+ * standard Hamiltonian and momentum constraints. The constraint-preserving
+ * portion of the boundary conditions is designed to prevent the influx of
+ * constraint violations from external faces of the evolution domain, by damping
+ * them away on a controlled and short time-scale. These conditions are imposed
+ * as corrections to the projections of the right-hand-sides of the GH evolution
+ * equations (i.e. using Bjorhus' method \cite Bjorhus1995), and are
+ * written down in Eq. (63) - (65) of \cite Lindblom2005qh . The gauge degrees
+ * of freedom are controlled by imposing a Sommerfeld-type condition (\f$L=0\f$
+ * member of the hierarchy derived in \cite BaylissTurkel) that allow gauge
+ * perturvations to pass through the boundary without strong reflections. These
+ * assume a spherical outer boundary, and can be written down as in Eq. (25) of
+ * \cite Rinne2007ui . Finally, the physical boundary conditions control the
+ * influx of inward propagating gravitational-wave solutions from the external
+ * boundaries. These are derived by considering the evolution system of the Weyl
+ * curvature tensor, and controlling the inward propagating characteristics of
+ * the system that are proportional to the Newman-Penrose curvature spinor
+ * components \f$\Psi_4\f$ and \f$\Psi_0\f$. Here we use Eq. (68) of
+ * \cite Lindblom2005qh to only disallow any incoming waves.
+ *
```

Comment:
Should it be stated that for char speed precisely equal to zero, that the BC is imposed?

Reply:
sure, added a sentence to that effect.